### DISCUSSION TITLE
Subdirectories in plugins in Airflow 2.0 are not recognized any more

### DISCUSSION THREAD

**Discussion by luebbert42** [apache/airflow#15678] [General]
I have a couple of Airflow 1.x plugins which do not work anymore in Airflow 2.x.
Airflow 2.x does not find files in subdirectories e.g. plugins/my_plugin/views/my_view.py so I get ModuleNotFoundErrors when starting airflow.

I have to append each subdirectory to sys.path to make the plugins work again:

```python 
  current_path = os.path.dirname(os.path.abspath(__file__))
  sys.path.insert(0,os.path.join(current_path, "views"))
  sys.path.insert(0, os.path.join(current_path, "models"))
```

This feels a bit weird. 
Is this the correct idea to solve my migration issue?

**Comment by potiuk** [comment]
Just out of curiosity (I am not sure what is the right answer it but I am happy to look more into some of the import problems). Do you have __init__.py files in your sub-folders? Would it be possible to share some of those migrated plugins so that I could try to reproduce it and maybe figure out the reason ?

**Reply by luebbert42** [reply]
@potiuk  - thanks for looking at my question :-) 

I think I have to migrate the plugin to a package ("pip install ...") to get it up and running in Airflow 2.x


https://github.com/luebbert42/importstruggle/blob/main/plugins/boom_plugin/boom_plugin.py

Anyway here's what I am trying to achieve, line 9 basically shows the problem. This line was working in Airflow 1.x and I cannot find out how make it work in Airflow 2.x. I only managed to add all subdirectories to the path and then I can import the files in the subdirectories again BUT I loose the namespace (which is ugly). 

I still think that I am doing something very wrong, because this is such a fundamental issue and there should be lots of bug reports on this, but there are not.

**Reply by potiuk** [reply]
No time to look at it so far. Anything changed/was fixed?

**Reply by luebbert42** [reply]
I gave up and converted it into a proper python package ("pip install...")
